openwisp_controller/config/api/filters.py

```diff
@@ -0,0 +1,100 @@
+from django_filters import rest_framework as filters
+from swapper import load_model
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+DeviceGroup = load_model('config', 'DeviceGroup')
+
+
+class BaseConfigAPIFilter(filters.FilterSet):
+    organization = filters.CharFilter(field_name='organization', label='Organization')
+
+    def _set_valid_filterform_lables(self):
+        # When not filtering on a model field, an error message
+        # with the label "[invalid_name]" will be displayed in filter form.
+        # To avoid this error, you need to provide the label explicitly.
+        raise NotImplementedError
+
+
+class TemplateListFilter(BaseConfigAPIFilter):
+    class Meta:
+        model = Template
+        fields = {
+            'organization': ['exact'],
+            'backend': ['exact'],
+            'type': ['exact'],
+            'default': ['exact'],
+            'required': ['exact'],
+            'created': ['exact', 'gte', 'lt'],
+        }
+
+
+class VPNListFilter(BaseConfigAPIFilter):
+
+    subnet = filters.CharFilter(field_name='subnet', label='VPN Subnet')
+
+    class Meta:
+        model = Vpn
+        fields = {
+            'backend': ['exact'],
+            'subnet': ['exact'],
+            'organization': ['exact'],
+        }
+
+
+class DeviceListFilter(BaseConfigAPIFilter):
+    config__templates = filters.CharFilter(
+        field_name='config__templates', label='Config template'
```

Comment:
these labels too, all strings which are shown in the UI have to be flagged as translatable